File: Daily Test/C/Combine Two Matrices - Vertically 19.1.2026.c
#include <stdio.h>

int main() {
    int r, c;
    scanf("%d %d", &r, &c);

    int A[r][c], B[r][c];
    for(int i=0;i<r;i++)
        for(int j=0;j<c;j++)
            scanf("%d",&A[i][j]);

    for(int i=0;i<r;i++)
        for(int j=0;j<c;j++)
            scanf("%d",&B[i][j]);

    int match = -1, kfirst = 0;

    for(int k=0;k<c;k++){
        int ok = 1;
        for(int i=0;i<r;i++)
            if(A[i][k] != B[i][0]) ok = 0;
        if(ok){ match = k; kfirst = 0; break; }
    }

    for(int k=0;k<c;k++){
        int ok = 1;
        for(int i=0;i<r;i++)
            if(A[i][k] != B[i][c-1]) ok = 0;
        if(ok){ match = k; kfirst = c-1; break; }
    }

    int s = (kfirst == 0) ? match : match - (c - 1);
    int mn = s < 0 ? s : 0;
    int mx = (s + c - 1 > c - 1) ? s + c - 1 : c - 1;
    int w = mx - mn + 1;

    int res[r][w];
    for(int i=0;i<r;i++)
        for(int j=0;j<w;j++)
            res[i][j]=0;

    for(int i=0;i<r;i++){
        for(int j=0;j<c;j++){
            res[i][-mn + j] += A[i][j];
            res[i][s - mn + j] += B[i][j];
        }
    }

    for(int i=0;i<r;i++){
        for(int j=0;j<w;j++)
            printf("%d ",res[i][j]);
        printf("\n");
    }
}

File: Daily Test/Python/Combine Two Matrices - Vertically 19.1.2026.py
r, c = map(int, input().split())
A = [list(map(int, input().split())) for _ in range(r)]
B = [list(map(int, input().split())) for _ in range(r)]

match = None
kfirst = 0

for k in range(c):
    if all(A[i][k] == B[i][0] for i in range(r)):
        match, kfirst = k, 0
        break

for k in range(c):
    if all(A[i][k] == B[i][c-1] for i in range(r)):
        match, kfirst = k, c-1
        break

s = match if kfirst == 0 else match - (c - 1)

mn = min(0, s)
mx = max(c - 1, s + c - 1)
w = mx - mn + 1

offA = -mn
offB = s - mn

res = [[0]*w for _ in range(r)]

for i in range(r):
    for j in range(c):
        res[i][offA + j] += A[i][j]
        res[i][offB + j] += B[i][j]

for row in res:
    print(*row)